fsspec/tests/test_api.py

```diff
@@ -57,6 +57,19 @@ def test_get_put(tmpdir):
     fs.put(tmpdir, "/more", recursive=True)
     assert fs.find("/more") == ["/more/dir/two", "/more/one", "/more/three"]
 
+    entries = ["/dir/two", "/one", "/three"]
```

Comment:
I agree with fixing the memory FS;
but perhaps the best way to show that we indeed do exactly what is expected, it might be best to use another temporary directory and make a number of tests of (inpath, outpath) and actually call "cp -r {inpath} {outpath}" (with the shell, on posix only) to show that the right thing happens.